src/pages/resultsView/ResultsViewPage.tsx

```diff
@@ -642,84 +642,94 @@ export default class ResultsViewPage extends React.Component<
+                                                    }
+                                                    inputGeneQuery={
+                                                        this.quickOQLQueryStore!
+                                                            .geneQuery
+                                                    }
+                                                    callback={(...args) => {
+                                                        this.quickOQLQueryStore!.geneQuery =
+                                                            args[2];
+                                                    }}
+                                                    location={
+                                                        GeneBoxType.DEFAULT
+                                                    }
+                                                    submitButton={
+                                                        this
+                                                            .quickOQLSubmitButton
+                                                    }
+                                                    error={
+                                                        this.quickOQLQueryStore!
+                                                            .submitError
+                                                    }
+                                                    messages={
+                                                        this.quickOQLQueryStore!
+                                                            .oqlMessages
+                                                    }
+                                                />
+                                            </div>
+                                        )}
+                                    </div>
+                                    {// we don't show the result tabs if we don't have valid query
+                                    this.showTabs &&
+                                        !this.resultsViewPageStore
+                                            .genesInvalid &&
+                                        !this.resultsViewPageStore
+                                            .isQueryInvalid && (
+                                            <MSKTabs
+                                                key={this.urlWrapper.hash}
+                                                activeTabId={
+                                                    this.resultsViewPageStore
+                                                        .tabId
                                                 }
-                                                messages={
-                                                    this.quickOQLQueryStore!
-                                                        .oqlMessages
+                                                unmountOnHide={false}
+                                                onTabClick={(id: string) =>
```

Comment:
`onTabClick={this.handleTabChange}` (make sure `handleTabChange` has `@autobind`)

Reply:
This is also missing in master branch. I added @autobind to _handleTabChange_ as per your suggestion.